core/src/engine.cpp

```diff
@@ -0,0 +1,34 @@
+
+#include "knoting/engine.h"
+
+namespace knot {
+
+Engine::Engine() {
+    m_windowModule = std::make_shared<knot::Window>(m_windowWidth, m_windowHeight, m_windowTitle);
+
+    m_engineModules.emplace_back(m_windowModule);
+
+    for (auto& module : m_engineModules) {
+        module->on_awake();
+    }
+}
+
+void Engine::update_modules() {
+    for (auto& module : m_engineModules) {
+        module->on_update(0);
+    }
+
+    m_windowModule->draw();
+
+    for (auto& module : m_engineModules) {
```

Comment:
Same as for the draw call

Reply:
I think I covered this in the conversation above but if I have missed something let me know.
this is here just so the state I left the project in is the same as when i started working eg still renderering block background